src/statsmodel.jl

```diff
@@ -106,6 +106,71 @@ function formula end
 formula(m::TableStatisticalModel) = m.mf.f
 formula(m::TableRegressionModel) = m.mf.f
 
+"""
+    termnames(model::StatisticalModel)
+
+Return the names of terms used in the formula of `model`.
+
+This is a convenience method for `termnames(formula(model))`.
+
+For `RegressionModel`s with only continuous predictors, this is the same as
+`(responsename(model), coefnames(model))`.
+
+For models with categorical predictors, the returned names reflect
+the variable name and not the coefficients resulting from
+the choice of contrast coding.
+"""
+termnames(model::StatisticalModel) = termnames(formula(model))
+"""
+    termnames(t::FormulaTerm)
+
+Return a two-tuple of `termnames` applied to the left and
+right hand sides of the formula.
+ 
+Note that until `apply_schema` has been called, literal `1` in formulae
+is interpreted as `ConstantTerm(1)` and will thus appear as `"1"` in the 
+returned term names.
+
+```jldoctest
+julia> termnames(@formula(y ~ 1 + x * y + (1+x|g)))
+("y", ["1", "x", "y", "x & y", "(1 + x) | g"])
+```
+
+Similarly, formulae with an implicit intercept will not have a `"1"`
+in their term names, because the implicit intercept does not exist until 
+`apply_schema` is called (and may not exist for certain model contexts).
+
+```jldoctest
+julia> termnames(@formula(y ~ x * y + (1+x|g)))
+("y", ["x", "y", "x & y", "(1 + x) | g"])
+```
+"""
+termnames(t::FormulaTerm) = (termnames(t.lhs), termnames(t.rhs))
+
+"""
+    termnames(term::AbstractTerm)
+
+Return the name(s) of column(s) generated by a term.  
```

Comment:
This isn't true for categorical terms, is it?

Reply:
copy-paste mistake -- this is very lightly adapted from the code for `coefnames`.